### PR TITLE
fix `Reloading a top page through submitting a form with 'target="_parent"' adds the iframe flag`

### PR DESCRIPTION
**Pull request by Farfurix**
https://github.com/DevExpress/testcafe-hammerhead/issues/1680

### Changes
1. Pass the actual `processedContext` parameter to `processHtml` (`document.write`, `Element.insertAdjacentHTML`, `Element.outerHTML` setter, `Range.createContextualFragment`).

### CODE REVIEW

**Comment by Farfurix** [comment]
@testcafe-build-bot retest

**Comment by Farfurix** [comment]
@testcafe-build-bot retest

File: src/client/sandbox/node/window.js [hunk 1]
```diff
@@ -1097,15 +1097,19 @@ export default class WindowSandbox extends SandboxBase {
...
-                DOMMutationTracker.onElementChanged(this);
+                DOMMutationTracker.onElementChanged(el);
 
                 if (parentEl && value !== null && value !== void 0) {
                     const parentDocument = findDocument(parentEl);
                     const parentWindow   = parentDocument ? parentDocument.defaultView : null;
 
-                    nativeMethods.elementOuterHTMLSetter.call(this, processHtml(String(value), { parentTag: parentEl.tagName }));
+                    nativeMethods.elementOuterHTMLSetter.call(el, processHtml(String(value), {
+                        parentTag:        parentEl.tagName,
```
Comment (LavrovArtem) [inline src/client/sandbox/node/window.js]: ```parentEl && parentEl.tagName```

File: test/client/fixtures/sandbox/iframe-flag-test.js [hunk 2]
```diff
@@ -360,6 +360,57 @@ test('change a url via location from cross-domain window', function () {
...
+module('"processHTML" with actual context');
+
+test('"document.write" (GH-1680)', function () {
+    return createTestIframe()
+        .then(function (iframe) {
+            var iframeDocument = iframe.contentDocument;
+            var HTMLText       = '<form action="some-path.html" target="_parent"></form>';
+
+            iframeDocument.open();
+            iframeDocument.write(HTMLText);
```
Comment (LavrovArtem) [inline test/client/fixtures/sandbox/iframe-flag-test.js]: ```iframeDocument.write('<form action="some-path.html" target="_parent"></form>');```

File: test/client/fixtures/sandbox/iframe-flag-test.js [hunk 3]
```diff
@@ -360,6 +360,57 @@ test('change a url via location from cross-domain window', function () {
...
+});
+
+test('"outerHTML" setter (GH-1680)', function () {
+    var parentDiv = document.createElement('div');
+    var childDiv  = parentDiv.appendChild(document.createElement('div'));
+    var htmlText  = '<form action="some-path.html" target="_parent"></form>';
+
+    document.body.appendChild(parentDiv);
+
+    childDiv.outerHTML = htmlText;
```
Comment (LavrovArtem) [inline test/client/fixtures/sandbox/iframe-flag-test.js]: ```childDiv.outerHTML = '<form action="some-path.html" target="_parent"></form>';```

**Pull request merged**